### PR TITLE
Move to CMake targets

### PR DESCRIPTION
**Pull request by awegrzyn**

### CODE REVIEW

**Comment by awegrzyn** [comment]
@kostorr @pippohub @Barthelemy

**Comment by Barthelemy** [comment]
looks good, need more time to review it properly

**Comment by kostorr** [comment]
PDA dependencies don't get built on slc7

**Comment by awegrzyn** [comment]
What do you mean by PDA dependencies? You mean classes that depend on PDA?

**Comment by kostorr** [comment]
Yes and consequently the relevant executables. Those that support dummy interfaces only run with them. 

Executables also cannot be found in `PATH` after running `alienv enter ReadoutCard`.

`sw/slc7_x86_64/ReadoutCard/latest/bin` only contains `roc-setup-hugetlbfs`.
`sw/BUILD/ReadoutCard-latest/ReadoutCard/bin` contains all the executables - but built for the dummy interfaces.

**Comment by awegrzyn** [comment]
@kostorr You can give it a try (although there still might be some glitches)

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,72 +4,83 @@
...
-        CMP0025 # CMake 3.0
-        CMP0042 # CMake 3.0
-        )
-    IF (POLICY ${p})
-        cmake_policy(SET ${p} NEW)
-    ENDIF ()
-endforeach ()
+# Define project
+project(ReadoutCard
+  VERSION 0.0.0
```
Comment (Barthelemy) [inline CMakeLists.txt]: 0.9.2

File: CMakeLists.txt [hunk 2]
```diff
@@ -220,21 +262,93 @@ set(TEST_SRCS
...
+list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" isSystemDir)
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  if("${isSystemDir}" STREQUAL "-1")
+    set(CMAKE_INSTALL_RPATH "@loader_path/../${CMAKE_INSTALL_LIBDIR}")
+  endif()
+  set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+endif()
+unset(isSystemDir)
+
+# Install library
```
Comment (Barthelemy) [inline CMakeLists.txt]: `# Install library and executables`

**Comment by kostorr** [comment]
Everything works for me as well.

**Pull request merged**